Fix rate limit on health endpoint

dvx76 opened the pull request:
Only fetch the health endpoint (warning-lights) every 24h.

Fixes https://github.com/skodaconnect/homeassistant-myskoda/issues/468

This is just a proposal to allow some discussion. Would depend on updating `myskoda.get_vehicle()` to support an optional capability exclusion list, e.g.

```python
    async def get_vehicle(
            self, vin: str, excluded_capabilities: list[CapabilityId] | None
    ) -> Vehicle:
        """Load a full vehicle based on its capabilities."""
        capabilities = [
            CapabilityId.AIR_CONDITIONING,
            CapabilityId.AUXILIARY_HEATING,
            CapabilityId.CHARGING,
            CapabilityId.PARKING_POSITION,
            CapabilityId.STATE,
            CapabilityId.TRIP_STATISTICS,
            CapabilityId.VEHICLE_HEALTH_INSPECTION,
            CapabilityId.DEPARTURE_TIMERS,
        ]

        if excluded_capabilities:
            capabilities = [c for c in capabilities if c not in excluded_capabilities]

        return await self.get_partial_vehicle(vin, capabilities)
```

WebSpider commented:
I like this method of dealing with the cache, it allows for a central location to determine what we want to pull for this particular update.

I think in the PR itself there could be a few things done differently, but I understand it's just a proposal:
1. The 24h time period should be a constant
2. I think when reading the PR, right now the `VEHICLE_HEALTH_INSPECTION` gets excluded exactly at the time we want to include it: At cache expiry

dvx76 commented:
Will test after https://github.com/skodaconnect/myskoda/pull/367 is available.

WebSpider commented on custom_components/myskoda/coordinator.py:
```diff
@@ -133,30 +134,33 @@ async def _async_get_minimal_data(self) -> Vehicle:
-        else:
-            vehicle = await self.myskoda.get_vehicle(self.vin)
-        return vehicle
+                excluded_capabilities.append(CapabilityId.VEHICLE_HEALTH_INSPECTION)
+
```
Do we want to include `CapabilityId.VEHICLE_HEALTH_WARNINGS_WITH_WAKE_UP` here as well?

WebSpider replied:
Maybe add a debug log indicating cache is valid?

dvx76 replied:
`VEHICLE_HEALTH_WARNINGS_WITH_WAKE_UP` : didn't include it because it doesn't affect which endpoints are hit / isn't in the list of capabilities used by `MySkoda.get_vehicle()`

debug on valid cache: makes sense, adding it

WebSpider reviewed (changes requested):
PR looks fine in general to me, but for some reason `poetry` was bumped a major version. 
Let's make sure this is intentional and in some way limit the bandwidth `poetry` has to bump versions without us telling it to.

dvx76 commented on poetry.lock:
```diff
@@ -1,11 +1,12 @@
-# This file is automatically @generated by Poetry 1.8.5 and should not be changed by hand.
+# This file is automatically @generated by Poetry 2.0.1 and should not be changed by hand.
```
That is just the poetry version I have installed locally. This is the downside of poetry I guess. Poetry installs the environment, but isn't part _of_ the environment. In our workflows and Justfile (which I don't use) we don't pin the version either.

Relevant: https://github.com/python-poetry/poetry/issues/3316 + https://github.com/python-poetry/poetry/pull/9547

Perhaps (in a separate PR) we can add this in pyproject.toml

```
[tool.poetry]
requires-poetry = ">=2.0"
```

WebSpider reviewed (approved):
Agreed with the remarks. Let's go!

dvx76 commented:
I need to retest this after picking up the changes from https://github.com/skodaconnect/myskoda/pull/369 - right now excluding health will result in `Vehicle.health = None`, causing the next update to re-fetch health, and so on 😅

dvx76 commented:
This is now ready. Rebased, updated and tested:

Testing with poll interval 1min and cache validity manually forced to `minutes=5`

```
2025-03-07 14:17:18.552 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:17:18.553 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:17:24.781 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 6.229 seconds (success: True)

2025-03-07 14:18:24.439 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:18:24.440 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:18:24.440 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping health update - cache is still valid.
2025-03-07 14:18:32.718 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 8.279 seconds (success: True)

2025-03-07 14:19:32.441 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:19:32.441 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:19:32.441 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping health update - cache is still valid.
2025-03-07 14:19:41.604 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 9.163 seconds (success: True)

2025-03-07 14:20:41.441 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:20:41.441 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:20:41.441 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping health update - cache is still valid.
2025-03-07 14:20:52.097 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 10.656 seconds (success: True)

2025-03-07 14:21:52.445 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:21:52.446 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:21:52.446 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping health update - cache is still valid.
2025-03-07 14:21:57.773 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 5.328 seconds (success: True)

2025-03-07 14:22:57.443 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:22:57.444 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:22:57.444 DEBUG (MainThread) [custom_components.myskoda.coordinator] Updating health - cache expired at 2025-03-07 13:22:24.780531+00:00
2025-03-07 14:23:01.334 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 3.891 seconds (success: True)

2025-03-07 14:24:01.443 DEBUG (MainThread) [custom_components.myskoda.coordinator] Performing scheduled update of all data for vin TMBJXXXXXXXXXXXX
2025-03-07 14:24:01.443 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping user update - cache is still valid.
2025-03-07 14:24:01.443 DEBUG (MainThread) [custom_components.myskoda.coordinator] Skipping health update - cache is still valid.
```

WebSpider reviewed (approved):
lgtm
